components/src/dynamo/vllm/omni/stage_worker.py

```
@@ -418,10 +462,163 @@ def _load_processor(func_path: str | None) -> Any:
+    """Expose request-level multimodal payloads on the sole completion output."""
+    request_multimodal_output = getattr(engine_inputs, "multimodal_output", None)
+    if not request_multimodal_output:
+        return
+
+    outputs = _iter_completion_outputs(engine_inputs)
+    if len(outputs) != 1:
+        return
+
+    completion = outputs[0]
+    if not getattr(completion, "multimodal_output", None):
+        setattr(completion, "multimodal_output", request_multimodal_output)
```

Ruff (0.15.12)

[warning] Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)